src/libsystemd-network/sd-ndisc.c

```diff
@@ -233,6 +245,11 @@ static int ndisc_recv(sd_event_source *s, int fd, uint32_t revents, void *userda
                 return 0;
         }
 
+        if (in6_addr_equal(&packet->sender_address, &nd->link_local_addr)) {
```

Comment:
maybe don't do this check if link_local_addr is all zeroes?

Reply:
Such case is already covered by the above `if` condition.